Simplify the way yarn-install action ensures yarn.lock stays unchanged after `yarn install`

ascrazy opened the pull request:
### Background

I have faced a problem integrating yarn-install action into [tracker-front](https://github.com/toptal/tracker-front) where we have yarn.lock that gets all packages through registry.yarnpkg.com, so not a single mention of registry.npmjs.com is there, and this leads to `Change registry in yarn.lock file to npm Artifact Registry` step failing with an error (e.g. https://github.com/toptal/tracker-front/actions/runs/12527127099/job/34940394300?pr=3264)
<img src="https://github.com/user-attachments/assets/e0caf0cd-6e51-4566-864b-368eb2c8bfaf" width="250" />

Initially tried to hunt down and fix this specific problem, but debugging this quite complex bash script came to conclusion it's kind of overengineered and we can achieve the same behavior with much more simple and even more performant implementation. Thus this is my suggestion

### Description

The idea if existing implementation is that after running yarn install we process entire yarn.lock and undo any registry overrides that we've done previously so we can compare yarn.lock after yarn install to one in git index. 

The idea of updated implementation is that we capture checksum of the yarn.lock right before running `yarn install` (with overridden URLs) and compare it to checksum after yarn install

### How to test

- Check `yarn-install` works as expected on different toptal repos
- - success path - yarn-install passes when yarn.lock is up to date with package.json 
- - failure path - yarn-install fails when yarn.lock is to up to date with package.json 
- [x] toptal/staff-portal
- - success path - https://github.com/toptal/staff-portal/actions/runs/12668795818/job/35304940896?pr=14098#step:7:592
- - failure path - https://github.com/toptal/staff-portal/actions/runs/12668795818/job/35304940896?pr=14098#step:7:592
- [x] toptal/hire-global
- - success path - https://github.com/toptal/hire-global/actions/runs/12640219148/job/35220183956?pr=520#step:7:556
- - failure path - https://github.com/toptal/hire-global/actions/runs/12640053550/job/35219662580?pr=520#step:7:583

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/davinci-github-actions/blob/master/_docs/contribution/changeset-guidelines.md) (if needed)

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-anvil ping reviewers` - Ping teams for review

</details>

ascrazy commented:
Hi @sashuk,

I've noticed you were involved into testing of #295 when many of those bash scripts I'm modifying here were introduced, so could you please have a look at this PR and share what do you think of my suggestion? Maybe you know any pitfalls I might be missing?

pudek357 commented on yarn-install/action.yml:
```diff
@@ -112,34 +112,33 @@ runs:
     - name: Change registry in yarn.lock file to npm Artifact Registry
       if: "inputs.checkout-token && inputs.npm-gar-token && (contains(runner.name, 'inf-gha-runners') || contains(runner.name, 'ubuntu2204'))"
       shell: bash
+      working-directory: ${{ inputs.path }}
```
I was testing your changes [here](https://github.com/toptal/platform/actions/runs/12645855832/workflow), and it failed [here](https://github.com/toptal/platform/actions/runs/12645855832/job/35235614218#step:19:36) with error:
```
/home/runner/_work/_temp/fadd9d55-a1ca-47e0-8462-623ebd0f7f3d.sh: line 3: screening-wizard-frontend/yarn.lock.toptal: No such file or directory
grep: screening-wizard-frontend/yarn.lock: No such file or directory
```

pudek357 replied:
It passed here: https://github.com/toptal/platform/actions/runs/12646389199/job/35237031043
when I introduced those changes: https://github.com/toptal/platform/commit/d07eaa2b8b67adb388092e85d9af9f78d3e7e35c

It looks like we need to remove all occurrences of `${{ inputs.path }}` from this step.

ascrazy replied:
fixed. thanks a lot for testing it & catching the issue

sashuk commented on yarn-install/action.yml:
```diff
@@ -112,41 +112,32 @@ runs:
+        sed -i -e "s#https://registry.yarnpkg.com/#https://us-central1-npm.pkg.dev/toptal-ci/npm-registry/#g" yarn.lock
+        sed -i -e "s#https://registry.npmjs.org/#https://us-central1-npm.pkg.dev/toptal-ci/npm-registry/#g" yarn.lock
 
-        echo "Removing double quotes from the URLs"
```
Do you think we could keep the comments?

ascrazy replied:
Restored

sashuk commented on .changeset/fuzzy-bees-applaud.md:
```diff
@@ -0,0 +1,5 @@
+---
+"davinci-github-actions": major
```
Changes affect the way action works under the hood, right? If so, then maybe it is a patch as it is more like a refactoring?

ascrazy replied:
I was just afraid it might have some side-effects... But yeah, following semver semantics patch-level bump should be enough in this case

sashuk reviewed (approved):
Thank you for testing it in two apps, looks good!

ascrazy commented:
I have addressed the concerns raised by @sashuk and @pudek357 and re-rested on toptal/staff portal and on toptal/platform https://github.com/toptal/platform/actions/runs/12673183166/job/35318966291#step:11:624

pudek357 reviewed (approved):
Great job!
